landscape/client/deployment.py

```diff
@@ -208,14 +210,22 @@ def auto_configure(self):
         if not enabled or configured:
             return
 
-        title = generate_computer_title(auto_enroll_conf)
-        if title:
-            self.computer_title = title
-            self.write()
+        for _ in range(max_retries):
+            title = generate_computer_title(auto_enroll_conf)
+            if title:
+                self.computer_title = title
+                self.write()
 
-            auto_enroll_conf["configured"] = True
-            client_conf["auto-register"] = auto_enroll_conf
-            snap_http.set_conf("landscape-client", client_conf)
+                auto_enroll_conf["configured"] = True
+                client_conf["auto-register"] = auto_enroll_conf
+                snap_http.set_conf("landscape-client", client_conf)
+
+                break
+            elif retry:
```

Comment:
The way this is currently written, it will still retry even if `retry` is `False` and `title` is truthy, it just won't `time.sleep`. Is that the intended behaviour?

Reply:
That's not the intended behavior. I've refactored it, thanks.